src/sv-pipeline/04_variant_resolution/scripts/clean_vcf_part5_update_records.py

```diff
@@ -0,0 +1,185 @@
+#!/usr/bin/env python
+
+import argparse
+from collections import Counter
+import gzip
+import pysam
+import sys
+import svtk.utils as svu
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description=__doc__,
+        formatter_class=argparse.RawDescriptionHelpFormatter)
+    parser.add_argument('revise_vcf_lines', type=argparse.FileType('r'))
+    parser.add_argument('normal_revise_vcf')
+    parser.add_argument('famfile', type=argparse.FileType('r'))
+    parser.add_argument('sexchr_revise')
+    parser.add_argument('multi_geno_ids_txt')
+    parser.add_argument('outlier_samples_list', type=argparse.FileType('r'))
+    args = parser.parse_args()
+
+    # load the revised lines and index by ID
+    revised_lines_by_id = {}
+    with pysam.VariantFile(args.revise_vcf_lines) as revise_vcf:
+        header2 = revise_vcf.header
+        for record in revise_vcf:
+            revised_lines_by_id[record.id] = record
```

Comment:
```suggestion
    with pysam.VariantFile(args.revise_vcf_lines) as revise_vcf:
        header2 = revise_vcf.header
    revised_lines_by_id = {record.id: record for record in revise_vcf}
```
When they're possible comprehensions are generally faster than appends